### PR TITLE
refactor(kit): support noUncheckedIndexedAccess

### PR DESCRIPTION
**Pull request by splincode**

### CODE REVIEW

File: projects/kit/components/preview/preview.component.ts [hunk 1]
```diff
@@ -113,13 +113,11 @@ export class TuiPreviewComponent {
     }
 
     protected onResize(contentResizeEntries: readonly ResizeObserverEntry[]): void {
-        if (contentResizeEntries.length === 0) {
-            return;
-        }
-
-        const {width, height} = contentResizeEntries[0].contentRect;
+        const rect = contentResizeEntries[0]?.contentRect;
```
Comment (waterplea) [inline projects/kit/components/preview/preview.component.ts:116]: Let's destruct as `[entry]: ...` in the argument above

File: projects/kit/components/carousel/carousel.component.ts [hunk 2]
```diff
@@ -130,11 +130,12 @@ export class TuiCarouselComponent {
         return index < this.index || index > this.index + this.itemsCount;
     }
 
-    protected onIntersection(
-        {intersectionRatio}: IntersectionObserverEntry,
-        index: number,
-    ): void {
-        if (intersectionRatio && intersectionRatio !== 1 && !this.transitioned) {
+    protected onIntersection(entry?: IntersectionObserverEntry, index = 0): void {
```
Comment (waterplea) [inline projects/kit/components/carousel/carousel.component.ts:133]: Is it even possible to not have no entries here and still have a callback? Sounds like we are overcomplicating things here for no reason.

File: projects/kit/components/files/input-files/input-files.directive.ts [hunk 3]
```diff
@@ -54,11 +54,13 @@ export class TuiInputFilesDirective
...
 
     public process(files: FileList): void {
-        this.onChange(
-            this.input.multiple
-                ? [...toArray(this.value()), ...Array.from(files)]
-                : files[0] || null,
-        );
+        const fileOrFiles = this.input.multiple
+            ? [...toArray(this.value()), ...Array.from(files)]
+            : files[0] || null;
```
Comment (waterplea) [inline projects/kit/components/files/input-files/input-files.directive.ts:59]: No need to fallback to `null` if we check below

File: projects/kit/components/range/range-change.directive.ts [hunk 4]
```diff
@@ -56,7 +56,7 @@ export class TuiRangeChange {
                     }
                 }),
                 switchMap((event) => this.pointerMove$.pipe(startWith(event))),
-                map(({clientX}) => this.getFractionFromEvents(clientX)),
+                map((event) => this.getFractionFromEvents(event?.clientX ?? 0)),
```
Comment (waterplea) [inline projects/kit/components/range/range-change.directive.ts:59]: We is `event` optional?

File: projects/kit/components/tabs/tabs-with-more.component.ts [hunk 5]
```diff
@@ -231,20 +231,22 @@ export class TuiTabsWithMore implements AfterViewChecked, AfterViewInit {
 
         const {exposeActive, minMoreWidth} = this.options;
         const {clientWidth} = this.el;
-        const activeWidth = tabs[activeItemIndex] ? tabs[activeItemIndex].scrollWidth : 0;
-        const moreWidth = Math.max(tabs[tabs.length - 1].scrollWidth, minMoreWidth);
+        const activeWidth = tabs[activeItemIndex]
+            ? (tabs[activeItemIndex]?.scrollWidth ?? 0)
```
Comment (waterplea) [inline projects/kit/components/tabs/tabs-with-more.component.ts:235]: We just checked that it exists, doesn't Typescript understand that? Let's change to:
```ts
const active = tabs[activeItemIndex];
const activeWidth = active?.scrollWidth ?? 0;
```

**Review by waterplea — APPROVED** [review]

File: projects/kit/components/preview/preview.component.ts (hunk 1 above)
Comment (splincode) [reply projects/kit/components/preview/preview.component.ts]: fixed

File: projects/kit/components/carousel/carousel.component.ts (hunk 2 above)
Comment (splincode) [reply projects/kit/components/carousel/carousel.component.ts]: fixed

File: projects/kit/components/files/input-files/input-files.directive.ts (hunk 3 above)
Comment (splincode) [reply projects/kit/components/files/input-files/input-files.directive.ts]: fixed

File: projects/kit/components/range/range-change.directive.ts (hunk 4 above)
Comment (splincode) [reply projects/kit/components/range/range-change.directive.ts]: fixed

File: projects/kit/components/tabs/tabs-with-more.component.ts (hunk 5 above)
Comment (splincode) [reply projects/kit/components/tabs/tabs-with-more.component.ts]: fixed

**Pull request merged**